pkg/tools/mcp_tool.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tools
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/kubectl-ai/gollm"
+	"github.com/GoogleCloudPlatform/kubectl-ai/pkg/mcp"
+	"k8s.io/klog/v2"
+)
+
+var mcpManager *mcp.Manager
+
+// =============================================================================
+// Schema Conversion Functions (kubectl-ai specific)
+// =============================================================================
+
+// convertToolToGollm converts an MCP tool to gollm.FunctionDefinition with a simple schema
+func convertToolToGollm(mcpTool *mcp.Tool) (*gollm.FunctionDefinition, error) {
+	return &gollm.FunctionDefinition{
+		Name:        mcpTool.Name,
+		Description: mcpTool.Description,
+		Parameters: &gollm.Schema{
+			Type: gollm.TypeObject,
+			Properties: map[string]*gollm.Schema{
+				"arguments": {
+					Type:        gollm.TypeObject,
+					Description: "Arguments for the MCP tool",
+					Properties:  map[string]*gollm.Schema{},
+				},
+			},
+		},
+	}, nil
+}
+
+// =============================================================================
+// MCP Tool Implementation
+// =============================================================================
+
+// init function is kept minimal - MCP initialization is handled by --mcp-client flag
+func init() {
+	// MCP client initialization will be handled via explicit --mcp-client flag
+	// No automatic discovery in init() to ensure proper control flow
+}
```

Comment:
in the spirit of explicit control, let's use instances instead of globals. Might be better to get rid of this init block.